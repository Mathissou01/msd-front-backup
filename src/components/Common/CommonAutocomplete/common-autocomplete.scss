@use "src/styles/02-tools/tools.include-media" as im;
@use "src/styles/01-settings/settings.colors" as colors;

.c-CommonAutocomplete {
  &__Container {
    display: flex;
    flex-direction: column;
    justify-content: center;
    width: 100%;
  }

  &__Label {
    font-weight: 700;
  }

  &__Input {
    display: flex;
    flex-direction: row;
    gap: 4px;
    align-items: center;
    box-sizing: border-box;
    height: 48px;
    padding: 0 8px;
    background: colors.$white;
    border: 1px solid;
    border-color: colors.$blue-2;
    border-radius: 4px;
  }

  &__Selector {
    width: 100%;
    max-height: 232px;
    margin-top: 2px;
    padding: 4px;
    overflow-y: scroll;
    background: colors.$white;
    border: 2px solid colors.$blue-4;
    border-radius: 2px;
    box-shadow: 0 0 1px rgb(17 95 251 / 5%), 0 1px 3px rgb(17 95 251 / 10%);
    cursor: pointer;

    &::-webkit-scrollbar {
      width: 4px;
      background-color: colors.$white;
      border-radius: 15px;
    }

    &::-webkit-scrollbar-thumb {
      background-color: colors.$gray-500;
    }
  }

  &__Option {
    gap: 10px;
    width: 100%;
    height: 56px;
    padding: 16px;
    text-align: left;
    background: colors.$white;
    border-bottom: 1px solid;
    border-color: colors.$blue-4;

    &:hover {
      background: colors.$gray-200;
    }
  }
}
