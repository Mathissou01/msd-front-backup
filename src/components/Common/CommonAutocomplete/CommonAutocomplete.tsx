import React, { Dispatch, SetStateAction } from "react";
import "./common-autocomplete.scss";

interface CommonAutocompleteProps {
  value: string;
  handleChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
  debouncedValue: string;
  isVisibled: boolean;
  setIsVisibled: Dispatch<SetStateAction<boolean>>;
  handleClick: (option: string) => void;
  filteredOptions: string[];
  inputName: string;
  inputLabel?: string;
  inputPlaceholder?: string;
}

const CommonAutocomplete: React.FC<CommonAutocompleteProps> = ({
  value,
  handleChange,
  debouncedValue,
  isVisibled,
  setIsVisibled,
  handleClick,
  filteredOptions,
  inputName,
  inputLabel = "",
  inputPlaceholder = "",
}) => {
  return (
    <div className="c-CommonAutocomplete">
      <div className="c-CommonAutocomplete__Container">
        <label htmlFor={inputName} className="c-CommonAutocomplete__Label">
          {inputLabel}
        </label>
        <input
          type="text"
          className="c-CommonAutocomplete__Input"
          value={value}
          name={inputName}
          onChange={handleChange}
          onFocus={() => setIsVisibled(true)}
          placeholder={inputPlaceholder}
        />
        {debouncedValue && isVisibled && (
          <div className="c-CommonAutocomplete__Selector">
            {filteredOptions.length ? (
              filteredOptions.map((option: string) => (
                <button
                  className="c-CommonAutocomplete__Option"
                  key={option}
                  onClick={() => handleClick(option)}
                >
                  {option}
                </button>
              ))
            ) : (
              <div className="c-CommonAutocomplete__Option">Aucun r√©sultat</div>
            )}
          </div>
        )}
      </div>
    </div>
  );
};

export default CommonAutocomplete;
