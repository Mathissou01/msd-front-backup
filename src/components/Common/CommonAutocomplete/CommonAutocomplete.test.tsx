import { render } from "@testing-library/react";
import CommonAutocomplete from "./CommonAutocomplete";

const mock = {
  value: "common",
  handleChange: () => {
    console.log("");
  },
  debouncedValue: "common",
  isVisibled: true,
  setIsVisibled: () => {
    console.log("");
  },
  handleClick: () => {
    console.log("");
  },
  filteredOptions: ["common autocomplete 1", "common autocomplete 2"],
  inputName: "common autocomplete",
  inputLabel: "common autocomplete",
  inputPlaceholder: "common autocomplete",
};

describe("CommonAutocomplete", () => {
  it("renders", () => {
    const { container } = render(
      <CommonAutocomplete
        value={mock.value}
        handleChange={mock.handleChange}
        debouncedValue={mock.debouncedValue}
        isVisibled={mock.isVisibled}
        setIsVisibled={mock.setIsVisibled}
        handleClick={mock.handleClick}
        filteredOptions={mock.filteredOptions}
        inputName={mock.inputName}
        inputLabel={mock.inputLabel}
        inputPlaceholder={mock.inputPlaceholder}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
