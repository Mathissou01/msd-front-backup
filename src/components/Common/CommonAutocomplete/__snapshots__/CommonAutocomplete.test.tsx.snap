// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonAutocomplete renders 1`] = `
<div>
  <div
    class="c-CommonAutocomplete"
  >
    <div
      class="c-CommonAutocomplete__Container"
    >
      <label
        class="c-CommonAutocomplete__Label"
        for="common autocomplete"
      >
        common autocomplete
      </label>
      <input
        class="c-CommonAutocomplete__Input"
        name="common autocomplete"
        placeholder="common autocomplete"
        type="text"
        value="common"
      />
      <div
        class="c-CommonAutocomplete__Selector"
      >
        <button
          class="c-CommonAutocomplete__Option"
        >
          common autocomplete 1
        </button>
        <button
          class="c-CommonAutocomplete__Option"
        >
          common autocomplete 2
        </button>
      </div>
    </div>
  </div>
</div>
`;
